### ISSUE TITLE
Browser mod hangs IOS app 

### ISSUE THREAD

**Issue by shawwellpete** [thomasloven/hass-browser_mod#703]
My Home Assistant version: 2024.4.4 - also tested om 2024.2.0

**What I am doing:**
Adding an IOS device to browser mod

**What I expected to happen:**
I expected to select the register slider, have the ability to change the browser id and thereby register a new device

**What happened instead:**
IOS becomes unresponsive for a couple of minutes, seemingly stuck on browsermod, then the app resets to the usual home screen

**Minimal** steps to reproduce:
I have removed browsermod completely and resinstalled it (2.3.0) I also updated home assistant from 
2024.2.0 to 2024.4.4 then installed browsermod

I have tested two ipad pros and an iphone, all with the same problem

My laptop using chrome works as expected so seems to be an IOS app issue

**By replacing the space in the checkboxes ([ ]) with an X below, I indicate that I:**

- [ x] Understand that this is a channel for reporting bugs, not a support forum (https://community.home-assistant.io/).

- [ x] Have made sure I am using the latest version of the plugin.

- [ x] Have followed the troubleshooting steps of the "Common Problems" section of https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins.

- [ x] Understand that leaving one or more boxes unticked or failure to follow the template above may increase the time required to handle my bug-report, or cause it to be closed without further action.

**Comment by stonegray** [comment]
Can reproduce. t happens on desktop but is still usable.  To verify we're talking about the same issue, this only happens when you select browser mod from the sidebar yes?

Console shows hundreds of the following error:
```
[Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'this.hass.localize')
	performUpdate (app.Icb7AiDppIM.js:2:135745)
	(anonymous function) (app.Icb7AiDppIM.js:2:135258)
[Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'this.hass.localize')
	(anonymous function) (app.Icb7AiDppIM.js:2:135198)
```

**Comment by shawwellpete** [comment]
> Can reproduce. t happens on desktop but is still usable.  To verify we're talking about the same issue, this only happens when you select browser mod from the sidebar yes?
> 
> 
> 
> Console shows hundreds of the following error:
> 
> ```
> 
> [Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'this.hass.localize')
> 
> 	performUpdate (app.Icb7AiDppIM.js:2:135745)
> 
> 	(anonymous function) (app.Icb7AiDppIM.js:2:135258)
> 
> [Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 'this.hass.localize')
> 
> 	(anonymous function) (app.Icb7AiDppIM.js:2:135198)
> 
> ```
> 
> 
> 
> 

Sorry that's the only way I know how to select browser mod, what is the alternative?

**Comment by stonegray** [comment]
No alternative, just wanted to verify it's only after selecting browser mod to make sure we're talking about the same bug. Wasn't clear what you meant by register slider. I can reproduce the hang on selecting it.

**Comment by stonegray** [comment]
Related #652

**Comment by shawwellpete** [comment]
> No alternative, just wanted to verify it's only after selecting browser mod to make sure we're talking about the same bug. Wasn't clear what you meant by register slider. I can reproduce the hang on selecting it.

Sorry if I wasn't here, by slider I meant this switch

![image](https://github.com/thomasloven/hass-browser_mod/assets/74064242/055459b6-6aef-4e3c-836a-1e58df6e1a90)

**Comment by shawwellpete** [comment]
I guess there is no update on this?
Anybody found a solution?

**Comment by scythk** [comment]
Same issue. When app opened it directly jumped into browser mod panel and stucked there, I was unable to open the siderbar and switch out

**Comment by krasnoukhov** [comment]
Maybe it would be helpful to some people here, I switched to [Bubble-Card Pop-up](https://github.com/Clooos/Bubble-Card?tab=readme-ov-file#pop-up) thing a while back, works pretty good for my cases

**Comment by dcapslock** [comment]
It is likely that this is fixed in versions > 2.3.3. I can't replicate thiswith iOS.

**Comment by shawwellpete** [comment]
I can confirm it has been fine recently, not sure which update fixed it

**Comment by ankorinek** [comment]
Environment:

Home Assistant Core: 2026.6
Browser Mod: latest via HACS
iPadOS: 16.7.15 (iPad6,12)
Tested with: HA Companion App and Safari (Add to Home Screen)

Symptoms:
The HA Companion App never appears in the browser-mod panel at all — only Safari sessions show up, confirming the companion app's WebView does not load Browser Mod's frontend JavaScript.
Safari sessions do appear in the panel and can be registered, but the connection is not maintained. The panel consistently shows "last connected X minutes ago" rather than a live connection, even with the screen on, auto-lock disabled, and the Safari tab in the foreground. Calling browser_mod.navigate with the correct browser ID and path has no effect.
What I tested:

Companion app: never appears in browser-mod panel
Safari (Add to Home Screen): connects briefly on page load, then immediately drops websocket
browser_mod.navigate called via Developer Tools → Actions immediately after connection shows in panel — no effect on device
Broadcasting without a browser_id (targeting all browsers) — no effect
Confirmed correct dashboard path from URL bar (non-standard path, not /lovelace/)
No errors in HA logs related to browser_mod

Impact:
This makes Browser Mod's navigate functionality completely unusable on iOS for dashboard automation use cases, which seems to be one of the primary use cases for wall-mounted tablets.
Question for maintainer:
Is the websocket drop a known iOS Safari limitation, or is there something in Browser Mod's connection handling that could be made more resilient to iOS's aggressive background tab throttling? Happy to test any diagnostic builds or provide additional information.

**Comment by dcapslock** [comment]
@ankorinek did you mean to submit a new issue? Locking this conversation now as commenting on closed issues is not best practice to get support.
